app/helpers/templateGenerator.js

```diff
@@ -1,119 +1,148 @@
       regex1: /&lt/g,
       string1: '<',
       regex2: /&gt;/g,
       string2: '>',
-      stringStart: ', ',
-      stringStart0: '',
-      accArray: ['{', '}'],
     },
     code: {
       regex1: /</g,
       string1: '&lt',
       regex2: />/g,
       string2: '&gt;',
-      stringStart: ',\n        ',
-      stringStart0: '        ',
-      accArray: ['{\n', '\n      }'],
     },
   };
-  const opts = options[type];
-
-  return Object.entries(params).reduce((acc, [key, value], i) => {
-    let modifiedValue;
-    if (component === 'view-data-bulletlist' && key === 'data') {
-      modifiedValue = (typeof value === 'string') ? value.split(',') : value;
-    } else {
-      modifiedValue = value.replace(opts.regex1, opts.string1).replace(opts.regex2, opts.string2);
-    }
-    const keyValueString = (i !== 0)
-      ? (`${opts.stringStart}${key}: ${JSON.stringify(modifiedValue)}`)
-      : (`${opts.stringStart0}${key}: ${JSON.stringify(modifiedValue)}`);
-    acc[0] += keyValueString;
-    return acc;
-  }, opts.accArray).join('');
+
+  const opts = options[isCode ? 'display' : 'code'];
+
+  let html = isCode ? '' : '<div class="bcc-c-code-nested">';
+
+  html += showKey ? `${isCode ? '' : '<label class="bcc-c-code-key-label bcc-u-code-secondary-color">'}"${key}": ${isCode ? '' : '</label>'}` : '';
+
+  if (typeof newValue === 'string') {
+    html += `${isCode ? '' : '<span class="bcc-c-code-editable-content bcc-u-code-primary-color">'}"${isCode ? '' : '<span contenteditable="true">'}`;
+    html += `${newValue.replace(opts.regex1, opts.string1).replace(opts.regex2, opts.string2)}`;
+    html += `${isCode ? '' : '</span>'}"${isCode ? '' : '</span>'}${isLast ? ' ' : ','}${isCode ? '' : '<br>'}`;
+  } else if (Array.isArray(newValue)) {
+    html += '[';
+    html += isCode ? '' : '<br><div class="bcc-c-code-json-array">';
+    html += newValue.map((val, index) => generateEditableJSON({
+      key: index,
+      value: val,
+      isLast: index + 1 === Object.keys(newValue).length,
+      showKey: false,
+      isCode,
+    })).join('');
+    html += isCode ? '' : '</div>';
+    html += ']';
+    html += isLast ? ' ' : ',';
+    html += isCode ? '' : '<br>';
+  } else {
+    html += '{';
+    html += isCode ? '' : '<br><div class="bcc-c-code-json-object">';
+    html += Object.entries(newValue).map(([k, v], index) => generateEditableJSON({
+      key: k,
+      value: v,
+      isLast: index + 1 === Object.keys(newValue).length,
+      isCode,
+    })).join('');
+    html += isCode ? '' : '</div>';
+    html += '}';
+    html += isLast ? ' ' : ',';
+    html += isCode ? '' : '<br>';
+  }
+
+
+  html += isCode ? '' : '</div>';
+
+  return html;
 };
 
-const getSettings = (component) => {
-  const settingsString = fs.readFileSync(`app/components/${component}/settings.json`, 'utf-8');
+const generateEditableCode = (params, isCode = false) => {
+  const keys = Object.keys(params);
+  return keys.map((key, index) => generateEditableJSON({
```

Comment:
think we should be able to do the following;
```
Object.enteries(params).map(([key, value], index) => ... )
```

Reply:
this was to avoid the linting issues of it becoming just the return statement therefore asking me to remove the curly braces around the function but this then made the line to long and caused that linting issue instead

Reply:
i can change from keys to entries though